test/storage_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+require 'ttnt/storage'
+
+class StorageTest < TTNT::TestCase
+  def setup
+    @storage_file = "#{@repo.workdir}/.ttnt"
+    FileUtils.rm_rf(@storage_file)
+
+    @section = 'test'
+    @storage = TTNT::Storage.new(@repo)
+    @data = { 'a' => 1, 'b' => 2 }
+  end
+
+  def test_read_storage
+    File.write(@storage_file, { @section => @data }.to_json)
+    assert_equal @data, @storage.read(@section)
+  end
+
+  def test_read_storage_from_history
+    @storage.write!(@section, @data)
+    git_commit_am('Add data to storage file')
+    sha = @repo.head.target_id
+    new_data = { 'c' => 3 }
+    @storage.write!(@section, new_data) # write to a file in working tree
+    history_storage = TTNT::Storage.new(@repo, sha)
+    assert_equal @data, history_storage.read(@section)
```

Comment:
Don't you think that something like:

~~~ruby
assert !history_storage.read(@section).key?('c')
~~~

show more the intent of the test ? I've no strong feeling about this one though.